core/controllers/profile.py

```diff
@@ -379,11 +379,19 @@ def get(self):
         user_data = user_takeout_object.user_data
         user_images = user_takeout_object.user_images
 
+        # Ensure that the exported data does not contain a user ID.
+        user_data_json_string = json.dumps(user_data)
+        if python_utils.UNICODE(self.user_id) in user_data_json_string:
+            user_data_json_string = (
+                'There was an error while exporting ' +
+                'data. Please contact admin@oppia.org to export your data.')
+            user_images = []
```

Comment:
Hmm, should we check only for the ID of the user all for all possible IDs? I would probably prefer to check if `user_data_json_string` does not contain any `USER_ID_REGEX`

Reply:
@vojtechjelinek to clarify, do you mean that we should use some generic regex for IDs? Recall that we do export some model ids (since we use model ids as UIDs for some dictionary keys. I have previously had a discussion w/ Sean and Ben, and we concluded that it's OK if expose model IDs, as long as they are not user IDs). 

If you mean that we should have a generic regex for any user ID, then that sounds fine to me. I think the one follow up question I have about that is, what is the format of the user ID? Where can I find that information?

Reply:
@vojtechjelinek friendly ping :)

Reply:
Sorry, missed this, we have `USER_ID_REGEX` in feconf.py :slightly_smiling_face:

Reply:
Also, we have it in prod_validators.py and base_model_validators.py too, can you get rid of it at these places and only use the one from feconf.py everywhere.

Reply:
Got it! Done.